src/HubManager.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry;
+
+use Sentry\State\Hub;
+use Sentry\State\HubInterface;
+
+final class HubManager implements HubManagerInterface
+{
+    /**
+     * @var HubInterface|null The current hub
+     */
+    private static $currentHub;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCurrentHub(): HubInterface
```

Comment:
If it acts as a singleton I guess the class can be completely static.

So only static method and mark the constructor as private.